spec/integ/matrix-client-event-timeline.spec.js

```diff
@@ -1,5 +1,5 @@
 import * as utils from "../test-utils/test-utils";
```

Comment:
The title and changelog of this PR is not every descriptive: "Fix issues with getEventTimeline and thread roots". It just describes the code and not what's being fixed.

Potential option:

> Fix events from main timeline getting mixed in threads

Reply:
But this does more than that, that's just one symptom it fixes, changelog entries should be relatively short even if that's vague

Reply:
What other symptoms does it fix? The other problems are all variations of that AFAICT.

The problem is that following the link to this PR from the changelog also doesn't have any of the extra details.

Reply:
> The problem is that following the link to this PR from the changelog also doesn't have any of the extra details either.

Except it does, in the canonical source, the issue it links to

![image](https://user-images.githubusercontent.com/2403652/173460252-726ba435-3d7a-4634-ac1b-a6f024b63ea2.png)

Reply:
Why doesn't "Fix events from main timeline getting mixed in threads" work to describe what this does then?

Reply:
Because it also stops other effects, like stops an unrelated timeline being returned if you call with a threaded-timelineset and an event id from a diff thread/main timeline

Reply:
The description and linked issue don't explain this. And my changelog example can be extended to something like:

> Fix events from main timeline getting mixed in threads and generally events being mixed across timelines.

Which is way more useful than `Fix issues with getEventTimeline and thread roots` where [thread roots are one of the edge cases that it actually doesn't fix](https://github.com/matrix-org/matrix-js-sdk/pull/2444#discussion_r893025870) anyway. And doesn't describe the main issue it fixes that everyone is running into in the first place.

I had to decode all of this information from the diff and when I did, I found bugs. The expectations and why were not spelled out in the code or description where we can easily compare how the code behaves to those expectations. One (maybe 2) symptoms are explain in the linked issue.

Reply:
> The description and linked issue don't explain this.

Nor do they need to given threads in the js-sdk is marked `@experimental`

> where https://github.com/matrix-org/matrix-js-sdk/pull/2444#discussion_r893025870 anyway.

This isn't true, it fixed the issue for event timeline sets accessing thread roots which caused the issue you saw, and created a regression around accessing thread roots from main timeline event timeline sets

Reply:
> Nor do they need to given threads in the js-sdk is marked `@experimental`

While compromises and known problems are probably part of an `@experimental` feature, seems like we shouldn't just drop the ball on all of it. I think this problem generally exists in most PR's though. And our current plan/goal for threads is to make this not experimental where this code is bound to live on.

We should so we can better maintain this. We're not going to remember this fresh context that I am driving for in 6 months when we find another regression. I understand that some of this code will change when we get backend changes to better support threads but that info will just make it easier to refactor this in the future as well to see that we're still maintaining all of the previous behavior plus the extra benefits.


> This isn't true, it fixed the issue for event timeline sets accessing thread roots which caused the issue you were hitting with main timeline events showing up in threads, and created a regression around accessing thread roots from main timeline event timeline sets

I see what you mean. This is an example of what we should be explaining though.